codalab/rest/bundles.py

```diff
@@ -691,8 +691,15 @@ def _fetch_bundle_contents_blob(uuid, path=''):
     else:
         response.set_header('Content-Disposition', 'attachment; filename="%s"' % filename)
     response.set_header('Target-Type', target_info['type'])
-    response.set_header('X-Codalab-Target-Size', target_info['size'])
-
+    if target_info['type'] == 'file':
+        size = target_info['size']
+    elif not path and bundle_name:
+        # return data_size if the user requests the actual bundle
+        size = local.model.get_bundle(target.bundle_uuid).metadata.data_size
```

Comment:
Let's use the `local.model.get_bundle_metadata` function instead.

Reply:
I have updated the code to use `get_bundle_metadata `.